src/compiler/emitter.ts

```diff
@@ -3541,10 +3541,10 @@ module ts {
                 write(" ");
                 emitToken(SyntaxKind.OpenBraceToken, endPos);

```

Comment:
Nit: can we have an emitCaseBlock function where you do all the appropriate work for that child node specifically?